MatrixSDKTests/MXClientInformationServiceUnitTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+@testable import MatrixSDK
+
+class MXClientInformationServiceUnitTests: XCTestCase {
+
+    func testUpdateData() {
+        MXSDKOptions.sharedInstance().enableNewClientInformationFeature = true
+
+        let mockDeviceId = "some_device_id"
+        let credentials = MXCredentials(homeServer: "", userId: "@userid:example.com", accessToken: "")
+        credentials.deviceId = mockDeviceId
+        guard let session = MXSession(matrixRestClient: MXRestClientStub(credentials: credentials)) else {
+            XCTFail("Failed to setup test conditions")
+            return
+        }
+
+        let service = MXClientInformationService(withSession: session)
+
+        let type = service.accountDataType(for: session)
+
+        // no client info before
+        let clientInfo = session.accountData.accountData(forEventType: type)
+        XCTAssertNil(clientInfo)
+
+        service.updateData()
+
+        // must be set after updateData
+        let updatedInfo1 = session.accountData.accountData(forEventType: type)
+        XCTAssertNotNil(updatedInfo1)
+
+        // try updating again
+        service.updateData()
+
+        // must not be changed
+        let updatedInfo2 = session.accountData.accountData(forEventType: type)
+        XCTAssertTrue(NSDictionary(dictionary: updatedInfo1!).isEqual(to: updatedInfo2))
+
+        session.close()
+    }
+
+    func testRemoveData() {
+        let mockDeviceId = "some_device_id"
+        let credentials = MXCredentials(homeServer: "", userId: "@userid:example.com", accessToken: "")
+        credentials.deviceId = mockDeviceId
+        guard let session = MXSession(matrixRestClient: MXRestClientStub(credentials: credentials)) else {
+            XCTFail("Failed to setup test conditions")
+            return
+        }
+
+        let service = MXClientInformationService(withSession: session)
```

Comment:
Could this code block be extracted into a method to avoid repetition (e.g. createSession)?

Reply:
I have a thought about it but didn't like it afterwards because we're using `mockDeviceId` and we would also need to check optionals again, so it's still boring.

Reply:
Sorry, what do you mean?

Reply:
Nevermind, updated.